### PR TITLE
Add porting guide to help new contributors

### PR DESCRIPTION
**Pull request by virtuald**
I'm sure you have a lot more thoughts @auscompgeek ... feel free to add them.

### CODE REVIEW

File: PORTING_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
+Guidelines for porting WPILib examples to Python
```
Comment (auscompgeek) [inline PORTING_GUIDE.md:1]: Should we name this `CONTRIBUTING.md`?

File: PORTING_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+* Filenames should always be all lowercase
+* Function names are camelCase
+* Class names start with a capital letter
+* Class method names are camelCase
+* Class member variables such as `m_name` should be `self.name` in Python
+* Protected/private methods/members can optionally be prefixed with `_`
+
+Misc conversion thoughts
+
+* Comparisons to null such as `foo == null` become `foo is None`
```
Comment (auscompgeek) [inline PORTING_GUIDE.md:44]: We should add a linter (e.g. flake8 or [ruff]) to enforce this.

[ruff]: https://github.com/charliermarsh/ruff

File: PORTING_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    motor = wpilib.Talon(1)
+
+    def __init__(self):
+        # OK: variable assigned to `self`
+        self.motor = wpilib.Talon(1)
+```
+
+---
+
+Import order doesn't really matter, but we prefer the following convention:
```
Comment (auscompgeek) [inline PORTING_GUIDE.md:74]: Would it be worth setting up `isort` (or ruff) to do this? I'd suggest [`shed`] to automatically configure isort, but I'm not sure it'd work with our directory structure.

If we do it in [pre-commit], [pre-commit.ci] can do it for contributors.

[`shed`]: https://github.com/Zac-HD/shed
[pre-commit]: https://pre-commit.com
[pre-commit.ci]: https://pre-commit.ci

Comment (virtuald) [reply PORTING_GUIDE.md:74]: If you want to do it, that's fine. I don't feel strongly enough about this guideline to enforce it -- but for those who read it, it's nice to keep in mind.

File: PORTING_GUIDE.md (hunk 2 above)
Comment (virtuald) [reply PORTING_GUIDE.md:44]: I don't know how the linter would know that the thing was originally == null.

File: PORTING_GUIDE.md (hunk 1 above)
Comment (virtuald) [reply PORTING_GUIDE.md:1]: I suppose that's true.

**Pull request merged**